### DISCUSSION TITLE
How to see the very first URI navigation in Dev Tools?

### DISCUSSION THREAD

**Discussion by ShaunLoganOracle** [MicrosoftEdge/WebView2Feedback#3927] [Q&A]
I am using Webview2 SDK 1.0.902.49, RT 96.0.1054.53 and WPF
To help with debugging, I added a call to: `MyWebView2Control.CoreWebView2.OpenDevToolsWindow():`
right before I set `MyWebView2Control.Source = "<some valid uri that causes a 302 redirect>";`
I do this in the event handler for the `Loaded` event for my WPF window.  All the other initialization for the WebView2 has already happened.

What I observe is that the DevTools window opens as expected, and the WebView2 control shows the expected content (the login page) but I don't see any evidence in the Network or Sources tab that the initial URI was fetched.  I expect to see a request for the URI that results in a 302/redirect to a login page.  I **do** see the login page activity (fetching images & js files etc.) in the Network tab.  When I submit that page, I also see that navigation in the DevToolsWindow.

Should I be able to see that initial URI in the Dev Tools window?

**Comment by jasonstephen15** [comment]
Hi @ShaunLoganOracle thanks for the question. Perhaps it's because the initial URI fetch is before you call the function to open devtools in `loaded`? https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/navigation-events 

As a workaround, you can intercept the NavigationStarting Event and postWebMessage if you want to see it in console. Please let me know if this doesn't answer your question, and I can have a dev follow up with you.

**Comment by ShaunLoganOracle** [comment]
Thanks @jasonstephen15 for the reply and suggestion.  
>perhaps it's because the initial URI fetch is before you call the function to open devtools

The operative code in my `Loaded` event handler is:
```
          await InitializeWebView2Async ();

          BrowserControl.CoreWebView2.OpenDevToolsWindow ();

          BrowserControl.Source = LoginUri;
```
Setting the `Source` property has been working fine (so I am confident that the initialization of the WV2 control is correct).
